### PR TITLE
Add flags to system message to control terminal graphics

### PR DESCRIPTION
**Pull request by erikcorry**

### CODE REVIEW

File: tools/mirror.toit [hunk 1]
```diff
@@ -478,26 +479,24 @@ class MallocReport extends Mirror:
...
     result.add   "└────────────────────────────────────────────────────────────────────────┘"
 
   usage_letters_ use/int fullness/int -> string:
-    symbols := ""
-    if use & MEMORY_PAGE_TOIT_ != 0: symbols += "🐱"
-    if use & MEMORY_PAGE_BUFFERS_ != 0: symbols = "B"
-    if use & MEMORY_PAGE_EXTERNAL_ != 0: symbols = "X"
-    if use & MEMORY_PAGE_TLS_ != 0: symbols = "W"
-    if use & MEMORY_PAGE_MISC_ != 0: symbols = "M"
+    symbols := []
```
Comment (floitsch) [inline tools/mirror.toit:491]: I don't understand this change.
Doesn't the call-site need to change as well?

File: tools/mirror.toit [hunk 2]
```diff
@@ -478,26 +479,24 @@ class MallocReport extends Mirror:
       result.add "│$(%2d k)k pages.  All pages are $(%2d k)k, even the ones that are shown wider       │"
       result.add "│ because they have many different allocations in them.                  │"
     else:
-      result.add "│Each line is a $(%2d k)k page.                                                      │"
-    result.add   "│   X  = External strings/bytearrays.        B  = Network buffers.       │"
-    result.add   "│   W  = TLS/crypto.                         M  = Misc. allocations.     │"
-    result.add   "│   🐱 = Toit managed heap.                  -- = Free page.             │"
+      result.add "│Each line is a $(%2d k)k page.                                                │"
```
[the remark this answers is not in the document]
Comment (erikcorry) [reply tools/mirror.toit:482]: `$(%2d k)` is 8 characters, but only produces two characters, so it needs to look 6 characters longer.

File: tools/mirror.toit [hunk 3]
```diff
@@ -478,26 +479,24 @@ class MallocReport extends Mirror:
...
     if terminal:
+      result.add "│   X  = External strings/bytearrays.        B  = Network buffers.       │"
+      result.add "│   W  = TLS/crypto.                         M  = Misc. allocations.     │"
+      result.add "│   🐱 = Toit managed heap.                  -- = Free page.             │"
       result.add "│        Fully allocated $scale Completely free page.  │"
```
[the remark this answers is not in the document]
Comment (erikcorry) [reply tools/mirror.toit:487]: Similar issue here.

File: tools/mirror.toit [hunk 4]
```diff
@@ -478,26 +479,24 @@ class MallocReport extends Mirror:
...
+    if use & MEMORY_PAGE_BUFFERS_ != 0: symbols.add "Network Buffers"
+    if use & MEMORY_PAGE_EXTERNAL_ != 0: symbols.add "External strings/bytearrays"
+    if use & MEMORY_PAGE_TLS_ != 0: symbols.add "TLS/Crypto"
+    if use & MEMORY_PAGE_MISC_ != 0: symbols.add "Misc"
     if fullness == 0:
-      symbols = "--"
-    while symbols.size < 2:
-      symbols += " "
-    return symbols
+      symbols = ["(Free)"]
```
[the remark this answers is not in the document]
Comment (erikcorry) [reply tools/mirror.toit:498]: done

File: tools/mirror.toit (hunk 1 above)
Comment (erikcorry) [reply tools/mirror.toit:491]: Renamed to make it more clear that it's only used for plain ASCII mode, where the exact layout is not so constrained.

File: tools/system_message.toit [hunk 5]
```diff
@@ -23,13 +23,13 @@ import host.pipe
 import .snapshot
 import .mirror as mirror
 
-handle_system_message encoded_system_message snapshot_content:
+handle_system_message encoded_system_message snapshot_content --force_pretty=false --force_plain=false:
```
[the remark this answers is not in the document]
Comment (erikcorry) [reply tools/system_message.toit:26]: Done

File: tools/system_message.toit [hunk 6]
```diff
@@ -49,6 +50,10 @@ main args:
             --short_help="The snapshot file of the program that produced the message",
         cli.OptionString "message" --short_name="m" --required
             --short_help="The base64-encoded message from the device",
+        cli.Flag "force_pretty"
```
Comment (kasperl) [inline tools/system_message.toit:53]: Nit: I strongly prefer flags to use dashes not underscores.

File: tools/system_message.toit [hunk 7]
```diff
@@ -49,6 +50,10 @@ main args:
...
+            --short_help="Force the report to use terminal graphics",
+        cli.Flag "force_plain"
```
Comment (kasperl) [inline tools/system_message.toit:55]: Nit: I strongly prefer flags to use dashes not underscores.

Comment (erikcorry) [reply tools/system_message.toit]: I fixed it, and in the corresponding Jaguar change.

We should change cli so that either is accepted.

**Review by floitsch — APPROVED** [review]
LGTM except for the esp-idf change.

**Pull request merged**